backend/dal/Helpers/Extensions/LeaseExtensions.cs

```diff
@@ -0,0 +1,118 @@
+using Microsoft.EntityFrameworkCore;
+using Pims.Core.Extensions;
+using System;
+using System.Linq;
+using System.Security.Claims;
+using Entity = Pims.Dal.Entities;
+
+namespace Pims.Dal.Helpers.Extensions
+{
+    /// <summary>
+    /// LeaseExtensions static class, provides extension methods for leases.
+    /// </summary>
+    public static class LeaseExtensions
+    {
+        /// <summary>
+        /// Generate an SQL statement for the specified 'user' and 'filter'.
+        /// </summary>
+        /// <param name="query"></param>
+        /// <param name="user"></param>
+        /// <param name="filter"></param>
+        /// <returns></returns>
+        private static IQueryable<Entity.Lease> GenerateCommonLeaseQuery(this IQueryable<Entity.Lease> query, ClaimsPrincipal user, Entity.Models.LeaseFilter filter)
+        {
+            filter.ThrowIfNull(nameof(filter));
+            filter.ThrowIfNull(nameof(user));
+
+            if (!String.IsNullOrWhiteSpace(filter.TenantName)) {
+                query = query.Where(l => l.Tenant != null && EF.Functions.Like(l.Tenant.Surname + ", " + l.Tenant.FirstName + ", " + l.Tenant.MiddleNames, $"%{filter.TenantName}%"));
+            }
+
+            if (!String.IsNullOrWhiteSpace(filter.PidOrPin))
+            {
+                var pidOrPinValue = filter.PidOrPin.Replace("-", "").Trim();
+                if (Int32.TryParse(pidOrPinValue, out int pidOrPin)) {
+                    query = query.Where(l => l.Properties.FirstOrDefault() != null && (l.Properties.FirstOrDefault().PID == pidOrPin || l.Properties.FirstOrDefault().PIN == pidOrPin));
+                }
+            }
+
+            if (!String.IsNullOrWhiteSpace(filter.LFileNo))
+                query = query.Where(l => EF.Functions.Like(l.LFileNo, $"%{filter.LFileNo}%"));
+
+            if (filter.Sort?.Any() == true)
+                query = query.OrderByProperty(filter.Sort);
+            else
+                query = query.OrderBy(l => l.LFileNo);
+
+            return query.Include(l => l.Properties)
```

Comment:
Properties and PropertiesManyToMany isn't a very performant query.  Why do you need both?  Depending on the requirements it may be better to manually lazy load properties.

Reply:
will get this when I merge the other PR.